src/operator/mkl/mkldnn_memory-inl.h

```diff
@@ -0,0 +1,206 @@
+#include "mkldnn.hpp"
+#include "mkldnn_base-inl.h"
+#define CHECK_MKL_BUFFER 0
+#if CHECK_MKL_BUFFER == 1
+#include "../../operator_common.h"
+#include "../../mshadow_op.h"
+#endif
+using namespace mkldnn;
+
+namespace mxnet {
+
+template <typename Dtype>
+struct MKLDNNMemoryDescriptorBase : public PrvMemDescr,
+ public std::enable_shared_from_this<MKLDNNMemoryDescriptorBase<Dtype> > {
+    MKLDNNMemoryDescriptorBase(std::shared_ptr<memory::primitive_desc> usr_memory_pd
+        , std::shared_ptr<memory::primitive_desc> prv_memory_pd);
+
+    ~MKLDNNMemoryDescriptorBase() {
+    }
+    std::shared_ptr<MKLDNNMemoryDescriptorBase<Dtype> > get_shared_ptr() {
+      return this->shared_from_this();
+    }
+    // ---- PrvMemDescr virtual functions -----
+    void allocate() {
+      if (_prv_memory == nullptr) {
+        _prv_memory = std::shared_ptr<memory>(new memory(*_prv_memory_pd));
+        _internal_ptr = reinterpret_cast<Dtype *>(_prv_memory->get_data_handle());
+        _internal_size = prv_size();
+      }
+    }
+    std::shared_ptr<memory>  get_prv_memory(bool usage_check = false) {
+      if (_prv_memory == nullptr) {
+        if (usage_check)
+          CHECK_EQ(usage_check, true) << "get null prv memory from";
+        allocate();
+      }
+      return _prv_memory;
+    }
+    inline bool conversion_needed() const {
+      if (!_prv_memory_pd_not_null)
+        return false;
+      if (!_usr_memory_pd_not_null)
+        return false;
+      if (*_usr_memory_pd != *_prv_memory_pd)
+        return true;
+      else
+        return false;
+    }
+
+    void set_prv_memory_pd(std::shared_ptr<memory::primitive_desc> memory_pd) {
+      _prv_memory_pd = memory_pd;
+      if (_prv_memory_pd)
+        _prv_memory_pd_not_null = true;
+    }
+
+    void set_usr_memory_pd(std::shared_ptr<memory::primitive_desc> memory_pd) {
+      _usr_memory_pd = memory_pd;
+      if (_usr_memory_pd)
+        _usr_memory_pd_not_null = true;
+    }
+
+    virtual void* prv_ptr(bool allocate_when_uninit = true) {
+      return _internal_ptr;
+    }
+    virtual size_t prv_size() { return _prv_memory_pd->get_size(); }
+    virtual size_t prv_count() { return prv_size() / sizeof(Dtype); }
+
+    virtual bool layout_compare(std::shared_ptr<PrvMemDescr> other);
+    virtual PrvDescrType get_descr_type() { return PRV_DESCR_MKLDNN; }
+
+    std::shared_ptr<memory::primitive_desc>  prv_memory_pd() const {
+        return _prv_memory_pd;
+    }
+    std::shared_ptr<memory::primitive_desc>  usr_memory_pd() const {
+        return _usr_memory_pd;
+    }
+
+    std::string name;  // for debugging purposes
+
+    void check_usr_with_prv_descriptors();
+    void set_prv_memory(std::shared_ptr<memory> memory) {
+        _prv_memory = memory;
+        if (_prv_memory == nullptr) {
+          _internal_ptr = reinterpret_cast<Dtype *>(_prv_memory->get_data_handle());
+          _internal_size = prv_size();
+        } else {
+          VLOG(1) << "Set NULL Prv Memory";
```

Comment:
When will this log be printed

Reply:
it seems we currently do not use this function, it is there for future use-cases. if the caller of this function fails in creating mkldnn memory, it tries to print debug msg.